juju/juju
# Feature/clusterrolebinding

## Description of change

add cluster scope role/rolebinding for podspecV2;

## QA steps

- deploy charms `without` `global` rbac scope;

```yaml
serviceAccount:
  automountServiceAccountToken: true
  rules:
  - apiGroups: [""]
    resources: ["pods"]
    verbs: ["get", "watch", "list"]
kubernetesResources:
  serviceAccounts:
    - name: rbac-foo
      automountServiceAccountToken: true
      rules:
        - apiGroups: [""]
          resources: ["pods"]
          verbs: ["get", "watch", "list"]
```

- check role and  role binding created;

```bash
mkubectl get sa,roles,rolebindings,clusterroles,clusterrolebindings -nt1

NAME                                  SECRETS   AGE
serviceaccount/mariadb-k8s            1         11s
serviceaccount/rbac-foo               1         11s

NAME                                                  AGE
role.rbac.authorization.k8s.io/mariadb-k8s            11s
role.rbac.authorization.k8s.io/rbac-foo               11s

NAME                                                         AGE
rolebinding.rbac.authorization.k8s.io/mariadb-k8s            11s
rolebinding.rbac.authorization.k8s.io/rbac-foo

```


- deploy charms with `global` rbac scope;

```yaml
serviceAccount:
  automountServiceAccountToken: true
  global: true
  rules:
  - apiGroups: [""]
    resources: ["pods"]
    verbs: ["get", "watch", "list"]
kubernetesResources:
  serviceAccounts:
    - name: rbac-foo
      global: true
      automountServiceAccountToken: true
      rules:
        - apiGroups: [""]
          resources: ["pods"]
          verbs: ["get", "watch", "list"]
```

- check `cluster` role and `cluster` role binding created;

```bash
$ mkubectl get sa,roles,rolebindings,clusterroles,clusterrolebindings -nt1

NAME                                  SECRETS   AGE
serviceaccount/mariadb-k8s            1         36s
serviceaccount/rbac-foo               1         36s

NAME                                                                       AGE
clusterrole.rbac.authorization.k8s.io/t1-mariadb-k8s                       36s
clusterrole.rbac.authorization.k8s.io/t1-rbac-foo                          36s


NAME                                                                  AGE
clusterrolebinding.rbac.authorization.k8s.io/t1-mariadb-k8s           36s
clusterrolebinding.rbac.authorization.k8s.io/t1-rbac-foo              36s
```


## Documentation changes

No

## Bug reference

None
